apps/client/lib/services/auth.service.ts

```diff
@@ -0,0 +1,74 @@
+import {ApiError} from '../api/client';
+import {authApi} from '../api/auth.api';
+import {Result} from '../types/result';
+import {CookieService} from './cookie.service';
+import {LocalStorageService} from './localStorage.service';
+
+interface ApiToken {
+  token: string;
+}
+
+export type AuthTokenCallback = (token: string | null) => void;
+
+export namespace AuthService {
+  export const USER_TOKEN_KEY: string = 'auth.token';
+  export const SIGN_IN_ROUTE: string = '/auth/sign-in';
+  export const ACCESS_DENIED_ROUTE: string = '/access-denied';
+
+  let authToken: string | null = null;
+
+  let authTokenListeners: AuthTokenCallback[] = [];
+
+  export function addAuthTokenListener(fn: AuthTokenCallback) {
+    authTokenListeners.push(fn);
+  }
+
+  export function removeAuthTokenListener(fn: AuthTokenCallback) {
+    authTokenListeners = authTokenListeners.filter((x) => x !== fn);
+  }
+
+  function emitAuthToken(token: string | null) {
+    authTokenListeners.forEach((fn) => fn(token));
+  }
+
+  export function createOtp(
+    identifier: string,
+  ): Promise<Result<void, ApiError>> {
+    return authApi.post<void>('/auth/otp', {
+      body: JSON.stringify({identifier}),
+    });
+  }
+
+  export async function createSession(
+    identifier: string,
+    token: string,
+  ): Promise<Result<string | null, ApiError>> {
+    const res = await authApi.post<ApiToken | null>('/auth/session', {
+      body: JSON.stringify({identifier, token}),
+    });
+    return Result.map(res, (x) => x?.token ?? null);
+  }
+
+  export function setAuthToken(jwt: string) {
+    authToken = jwt;
+    emitAuthToken(jwt);
+  }
+
+  export function getAuthToken(): string | null {
+    return authToken;
+  }
+
+  export function saveJwt(jwt: string) {
+    LocalStorageService.setItem(USER_TOKEN_KEY, jwt);
+    CookieService.set(USER_TOKEN_KEY, jwt);
+  }
```

Comment:
This is a bit of a hack. It stores the JWT in local storage for client-side use and also adds it as a cookie so server components can get access to it when making requests to the API service from the Nextjs server.